libcnb/src/layer/handling.rs

```diff
@@ -368,21 +368,19 @@ fn write_layer<M: Serialize, P: AsRef<Path>>(
     layer_exec_d_programs: ExecDPrograms,
     layer_sboms: Sboms,
 ) -> Result<(), WriteLayerError> {
-    write_layer_metadata(layers_dir.as_ref(), layer_name, layer_content_metadata)?;
+    let layers_dir = layers_dir.as_ref();
 
-    let layer_dir = layers_dir.as_ref().join(layer_name.as_str());
+    write_layer_metadata(layers_dir, layer_name, layer_content_metadata)?;
+
+    let layer_dir = layers_dir.join(layer_name.as_str());
```

Comment:
Variable names `layers_dir` and `layer_dir` are very close to my eyes. Here you're introducing a variable to use it only in the next function call below. I think it would be less confusing to inline all that logic and remove the variable.


```rust
    layer_env.write_to_layer_dir(layers_dir.join(layer_name.as_str()))?;
```

Alternatively, `write_to_layer_dir` would be safer if it took a path to the layers dir and a `LayerName` and did this concatenation for you.

Reply:
We have over 100 occurrences of `layer_dir` and `layers_dir` variables in this project. We actually disable a specific lint in `handling.rs` to allow for this: 

```
// This lint triggers when both layer_dir and layers_dir are present which are quite common.
#![allow(clippy::similar_names)]
```

Regardless of how severe we think this this issue is, it's a not a problem of this PR I would say. If we feel strongly about this naming we should address it holistically. Personally, I don't think having `layers_dir` and `layer_dir` is a problem, but I can see your argument of them being similar.

Reply:
> Regardless of how severe we think this this issue is, it's a not a problem of this PR I would say. If we feel strongly about this naming we should address it holistically. Personally, I don't think having layers_dir and layer_dir is a problem, but I can see your argument of them being similar.

When dealing with paths it's easy to confuse passing in a directory you need versus passing in the parent directory of the directory you need. The naming is a piece of the puzzle. My larger worry would be that the types don't protect us in this situation.

If you're fine with it as-is, I'm fine with it. I'm poking things and seeing how they move.